controller/Review.php

```diff
@@ -100,68 +104,88 @@ public function init(): void
 
         $params = $this->getPsrRequest()->getQueryParams();
 
-        if (isset($params['serviceCallId'])) {
-
-            /** @var DeliveryExecutionFinderService $finder */
-            $finder = $this->getServiceLocator()->get(DeliveryExecutionFinderService::SERVICE_ID);
-
-            $data = $dataBuilder->create()->build($params['serviceCallId'], $finder->getShowScoreOption($this->ltiSession->getLaunchData()));
+        try {
+            if (isset($params['serviceCallId'])) {
+                /** @var DeliveryExecutionFinderService $finder */
+                $finder = $this->getServiceLocator()->get(DeliveryExecutionFinderService::SERVICE_ID);
+                $this->checkPermissions($finder->findDeliveryExecution($this->ltiSession->getLaunchData()));
+                $data = $dataBuilder->create()->build(
+                    $params['serviceCallId'],
+                    $finder->getShowScoreOption($this->ltiSession->getLaunchData())
+                );
+            }
+            $this->returnJson($data ?? []);
```

Comment:
the default case is `[ ]` do you think that is correct ? 
also i am not aware of the contract for this endpoint do you think we should try to create a small opeapi doc ?

Reply:
As I see FE works like that. 
About openapi doc, it's not the Rest API, so I'm not sure that we should create such kinds of docs. (btw, it's a very good point, but in TAO we don't do that by default)